mscore/seq.cpp

```diff
@@ -1405,10 +1415,13 @@ void Seq::putEvent(const NPlayEvent& event, unsigned framePos)
             qDebug("bad channel value %d >= %d", channel, cs->midiMapping()->size());
             return;
             }
+
+      // dispatch midi messages to MIDI driver first before rendering with MuseScore's synth, in order to avoid additional unpredictable latency associated with MuseScore's synth
+      if (_driver != 0 && (preferences.useJackMidi || preferences.useAlsaAudio || preferences.usePortaudioAudio))
+            _driver->putEvent(event, framePos);
```

Comment:
Regarding my [above comment](https://github.com/musescore/MuseScore/pull/3170#issuecomment-302268162), I've discovered that I am able to resolve the problem by outputting the event to MIDI driver *BEFORE* rendering them with MuseScore's internal synth.  This avoids the unpredictable latency of musescore's internal synth (which will slow down when it has to process & render a lot of notes in the same segment).  So now I'm able to listen to large 50-instrument scores via Microsoft's GM Wavetable synth without experiencing the jitter/lag I described earlier.